### ISSUE TITLE
[BUG] YouTubeChatDownloader().get_user_videos issue

### ISSUE THREAD

**Issue by HuskyTrojan** [xenova/chat-downloader#114]
## Basic information
- Program version:  0.1.7
- Python version: 3.7.11
- Operating system:  Windows 10

## Describe the bug
Not sure if this is a bug, just not erring out gracefully, a youtube issue, or something I'm doing wrong.  When looping through the generator created using the get_user_videos method in the YouTubeChatDownloader class after 30 iterations of the generator it fails to get any more videos.  I have tried this on multiple channels that contain more than 30 videos.

### Otherwise, if using the python module, provide the following:

1. A [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example):
```python

from chat_downloader.sites import YouTubeChatDownloader

for video in YouTubeChatDownloader().get_user_videos(custom_username='overwatchleague',video_type='past'):
    print(video)

```
2. Output, traceback or other information relating to the bug:
```
I've shown only the last of the 30 video results and then after that is what gets printed to the screen.  I've checked another channel with more than 30 videos and I get the same print out after the 30th video

{'video_id': '3wroAIisASE', 'title': 'Overwatch League 2021 Season | June Joust Tournament | Day 2', 'view_count': '', 'short_view_count': ''}
{'continuationItemRenderer': {'trigger': 'CONTINUATION_TRIGGER_ON_ITEM_SHOWN', 'continuationEndpoint': {'clickTrackingParams': 'CCQQ6IsCGAAiEwiEgdqtiIzyAhXDpMQKHYHQCm4=', 'commandMetadata': {'webCommandMetadata': {'sendPost': True, 'apiUrl': '/youtubei/v1/browse'}}, 'continuationCommand': {'token': '4qmFsgKDARIYVUNpQUluQkw5a1V6ejFYUnhrNjZ2LWd3GjhFZ1oyYVdSbGIzTVlBeUFDTUFFNEJPb0RGME5uUVZORGQybG9aM0pFUm1sSlF6Wm9aRGhDUzBSSpoCLGJyb3dzZS1mZWVkVUNpQUluQkw5a1V6ejFYUnhrNjZ2LWd3dmlkZW9zMTEw', 'request': 'CONTINUATION_REQUEST_TYPE_BROWSE'}}}}
```

## Expected behavior
I would expect this to continue until all videos have been found

## Screenshots
![image](https://user-images.githubusercontent.com/87031204/127723411-faef2e1b-e333-4133-a6d3-db166645e19b.png)

## Additional context/information
I'm a python novice, and my coding skills are limited to Matlab and Fortran.  So apologies up front if this is obvious to everyone else what is going on.

**Comment by xenova** [comment]
Hey there :wave:

Yes you are correct, I didn't actually finish implementing the `get_user_videos` method because I didn't need to get more than 30 videos. The only point in the code where the method is used is to check for current live streams (and I assumed a channel wouldn't have 30+ active live streams).

However, I can definitely finish implementing it :)

**Comment by xenova** [comment]
Alright, I have finished implementing it in this branch: https://github.com/xenova/chat-downloader/tree/improvements

You can install it using:
```
pip install git+https://github.com/xenova/chat-downloader.git@improvements
```

```python
from chat_downloader.sites import YouTubeChatDownloader

vids = YouTubeChatDownloader().get_user_videos(custom_username='overwatchleague', video_type='past')

count = 0
for video in vids:
    print(video)
    count += 1

print(count)
```

will output:
```
{'video_id': 'kWhSZeGJaDU', 'title': 'Overwatch League 2021 Season | Countdown Cup Qualifiers | Week 1 Day 1 — East', 'view_count': '153,165 views', 'short_view_count': '153K views'}
{'video_id': 'L_2pZH-oorQ', 'title': 'Overwatch League 2021 Season | Countdown Cup Qualifiers | Week 1 Day 1 — West', 'view_count': '298,127 views', 'short_view_count': '298K views'}
{'video_id': 'jXC5R8sHYVU', 'title': '[Co-Stream] Contenders North America | S2 July | Finals', 'view_count': '43,639 views', 'short_view_count': '43K views'}
{'video_id': 'ZjxjL7AZv8A', 'title': '[Co-Stream] Contenders Europe | S2 July | Finals', 'view_count': '48,542 views', 'short_view_count': '48K views'}
{'video_id': 'WZ_9Yx0od74', 'title': 'Grand Finals | Chengdu Hunters vs Shanghai Dragons | Summer Showdown Tournament', 'view_count': '288,810 views', 'short_view_count': '288K views'}
{'video_id': 'b3vanRpfNuU', 'title': 'Overwatch League 2021 Season | Summer Showdown Tournament | Day 2', 'view_count': '419,456 views', 'short_view_count': '419K views'}
{'video_id': 'rFvc2wXNzcE', 'title': 'Overwatch League 2021 Season | Summer Showdown Tournament | Day 1', 'view_count': '300,181 views', 'short_view_count': '300K views'}
{'video_id': '_mkMP2NyQoM', 'title': 'Overwatch League 2021 Season | Summer Showdown Qualifiers | Knockouts — West', 'view_count': '504,595 views', 'short_view_count': '504K views'}
{'video_id': 'XmXRBVKLSc8', 'title': 'Overwatch League 2021 Season | Summer Showdown Qualifiers | Knockouts — East', 'view_count': '232,020 views', 'short_view_count': '232K views'}
{'video_id': '_uRzfZjMSJQ', 'title': 'Overwatch League 2021 Season | Summer Showdown Qualifiers | Week 3 Day 3 — West', 'view_count': '471,047 views', 'short_view_count': '471K views'}
{'video_id': '_wN3wspBiQI', 'title': 'Overwatch League 2021 Season | Summer Showdown Qualifiers | Week 3 Day 2 — East', 'view_count': '230,420 views', 'short_view_count': '230K views'}
{'video_id': 'KIu7mKPff5I', 'title': 'Overwatch League 2021 Season | Summer Showdown Qualifiers | Week 3 Day 2 — West', 'view_count': '441,549 views', 'short_view_count': '441K views'}
{'video_id': 'Jd7jf0xmnNs', 'title': 'Overwatch League 2021 Season | Summer Showdown Qualifiers | Week 3 Day 1 — East', 'view_count': '278,618 views', 'short_view_count': '278K views'}
{'video_id': 'LjaKlIf9ZZY', 'title': 'Overwatch League 2021 Season | Summer Showdown Qualifiers | Week 3 Day 1 — West', 'view_count': '290,218 views', 'short_view_count': '290K views'}
{'video_id': 'xPaoMQpqYY0', 'title': 'Overwatch League 2021 Season | Summer Showdown Qualifiers | Week 2 Day 3 — West', 'view_count': '399,336 views', 'short_view_count': '399K views'}
{'video_id': 'HbhN923itks', 'title': 'Overwatch League 2021 Season | Summer Showdown Qualifiers | Week 2 Day 2 — East', 'view_count': '191,438 views', 'short_view_count': '191K views'}
{'video_id': 'S-dNCQJ135k', 'title': 'Overwatch League 2021 Season | Summer Showdown Qualifiers | Week 2 Day 2 — West', 'view_count': '438,131 views', 'short_view_count': '438K views'}
{'video_id': 'QYo436m6LZc', 'title': 'Overwatch League 2021 Season | Summer Showdown Qualifiers | Week 2 Day 1 — East', 'view_count': '153,668 views', 'short_view_count': '153K views'}
{'video_id': 'oojvVQ2Jatw', 'title': 'Overwatch League 2021 Season | Summer Showdown Qualifiers | Week 2 Day 1 — West', 'view_count': '301,072 views', 'short_view_count': '301K views'}
{'video_id': 'VbBC9IweMmg', 'title': 'Overwatch League 2021 Season | Summer Showdown Qualifiers | Week 1 Day 3 — West', 'view_count': '433,090 views', 'short_view_count': '433K views'}
{'video_id': '58t_39khLf8', 'title': 'Overwatch League 2021 Season | Summer Showdown Qualifiers | Week 1 Day 2 — East', 'view_count': '201,002 views', 'short_view_count': '201K views'}
{'video_id': 'tZvF3-HBFC0', 'title': 'Overwatch League 2021 Season | Summer Showdown Qualifiers | Week 1 Day 2 — West', 'view_count': '445,403 views', 'short_view_count': '445K views'}
{'video_id': 'mZhU8r4FT5Q', 'title': 'Overwatch League 2021 Season | Summer Showdown Qualifiers | Week 1 Day 1 — East', 'view_count': '215,033 views', 'short_view_count': '215K views'}
{'video_id': '0lK44okqw50', 'title': 'Overwatch League 2021 Season | Summer Showdown Qualifiers | Week 1 Day 1 — West', 'view_count': '404,482 views', 'short_view_count': '404K views'}
{'video_id': 'Rpk63AXNaGo', 'title': '[Co-Stream] Contenders China Season 1 | Losers Final & Grand Finals | Day 4', 'view_count': '22,271 views', 'short_view_count': '22K views'}
{'video_id': 'dRU7u5UWJFw', 'title': '[Co-Stream] Contenders China Season 1 | Playoffs | Day 3', 'view_count': '19,083 views', 'short_view_count': '19K views'}
{'video_id': 'deKJD7wyRhA', 'title': 'WORST. C9. EVER. 💩| Casters to Masters #3 — Bren (POV), Soe, Mr X, Sideshow, Uber, Danny', 'view_count': '49,765 views', 'short_view_count': '49K views'}
{'video_id': 'XGiiw4jL7j4', 'title': '[Co-Stream] Contenders North America | S2 June | Finals', 'view_count': '52,053 views', 'short_view_count': '52K views'}
{'video_id': 'qhk2ecK6Fu8', 'title': '[Co-Stream] Contenders Europe | S2 June | Finals', 'view_count': '70,306 views', 'short_view_count': '70K views'}
{'video_id': 'Nf8yOM1auzY', 'title': 'Overwatch League 2021 Season | June Joust Tournament Finals | Day 3', 'view_count': '382,344 views', 'short_view_count': '382K views'}
{'video_id': '3wroAIisASE', 'title': 'Overwatch League 2021 Season | June Joust Tournament | Day 2', 'view_count': '387,925 views', 'short_view_count': '387K views'}
{'video_id': '4S9lDEr_L0w', 'title': 'Overwatch League 2021 Season | June Joust Tournament | Day 1', 'view_count': '338,930 views', 'short_view_count': '338K views'}
{'video_id': 'lciX-91ebNE', 'title': 'Overwatch League 2021 Season | June Joust Qualifiers | Knockouts — West', 'view_count': '516,753 views', 'short_view_count': '516K views'}
{'video_id': 'MSx8YDVdysk', 'title': 'Overwatch League 2021 Season | June Joust Qualifiers | Week 3 Day 3 — Hangzhou Spark Homestand', 'view_count': '200,100 views', 'short_view_count': '200K views'}     
{'video_id': '8fEBVFL_cZ8', 'title': 'Overwatch League 2021 Season | June Joust Qualifiers | Week 3 Day 3 — West', 'view_count': '451,413 views', 'short_view_count': '451K views'}
{'video_id': 'hG2iKTKB_Kg', 'title': 'Overwatch League 2021 Season | June Joust Qualifiers | Week 3 Day 2 — Hangzhou Spark Homestand', 'view_count': '215,511 views', 'short_view_count': '215K views'}     
{'video_id': '8y7bfDwyZYA', 'title': 'Overwatch League 2021 Season | June Joust Qualifiers | Week 3 Day 2 — West', 'view_count': '396,133 views', 'short_view_count': '396K views'}
{'video_id': 'SoDqjO4_XIg', 'title': 'Overwatch League 2021 Season | June Joust Qualifiers | Week 3 Day 1 — Hangzhou Spark Homestand', 'view_count': '260,908 views', 'short_view_count': '260K views'}     
{'video_id': 'XV7NZQnQIk4', 'title': 'Overwatch League 2021 Season | June Joust Qualifiers | Week 3 Day 1 — West', 'view_count': '350,528 views', 'short_view_count': '350K views'}
{'video_id': 'jLaAGprUwGQ', 'title': 'Overwatch League 2021 Season | June Joust Qualifiers | Week 2 Day 3 — West', 'view_count': '478,775 views', 'short_view_count': '478K views'}
{'video_id': '6nChaIocgnY', 'title': 'Overwatch League 2021 Season | June Joust Qualifiers | Week 2 Day 2 — East', 'view_count': '165,265 views', 'short_view_count': '165K views'}
{'video_id': 'YIMpePYOP-M', 'title': 'Overwatch League 2021 Season | June Joust Qualifiers | Week 2 Day 2 — West', 'view_count': '417,486 views', 'short_view_count': '417K views'}
{'video_id': 'L_VBmykFp7w', 'title': 'Overwatch League 2021 Season | June Joust Qualifiers | Week 2 Day 1 — East', 'view_count': '180,301 views', 'short_view_count': '180K views'}
{'video_id': 'xZMh9XSNzN0', 'title': 'Overwatch League 2021 Season | June Joust Qualifiers | Week 2 Day 1 — West', 'view_count': '307,887 views', 'short_view_count': '307K views'}
{'video_id': '0KzKGR9wwp8', 'title': "'Winnable', Feeding, & A Broken Mouse... Casters to Masters #2 — Soe, Bren, Sideshow, Uber, Danny", 'view_count': '62,377 views', 'short_view_count': '62K views'}    
{'video_id': 'uamQv2_xWkA', 'title': 'Overwatch League 2021 Season | June Joust Qualifiers | Week 1 Day 3 — West', 'view_count': '412,482 views', 'short_view_count': '412K views'}
{'video_id': '7CrdNcSmm24', 'title': 'Overwatch League 2021 Season | June Joust Qualifiers | Week 1 Day 2 — East', 'view_count': '297,683 views', 'short_view_count': '297K views'}
{'video_id': 'yvDiOIZ29Bc', 'title': 'Overwatch League 2021 Season | June Joust Qualifiers | Week 1 Day 2 — West', 'view_count': '477,140 views', 'short_view_count': '477K views'}
{'video_id': 'ZMrk_yu2wKI', 'title': 'Overwatch League 2021 Season | June Joust Qualifiers | Week 1 Day 1 — East', 'view_count': '263,375 views', 'short_view_count': '263K views'}
{'video_id': 'CHJhv0PRW9M', 'title': 'Overwatch League 2021 Season | June Joust Qualifiers | Week 1 Day 1 — West', 'view_count': '372,169 views', 'short_view_count': '372K views'}
{'video_id': '5JM0LPrw6Fk', 'title': 'Overwatch 2 Developer PVP Livestream', 'view_count': '270,653 views', 'short_view_count': '270K views'}
{'video_id': 'S1m4Wkc2uhU', 'title': 'Here. We. GO... Casters to Masters #1 — Sideshow (POV), Soe, Uber, Mr X, Bren, & Danny Lim!', 'view_count': '103,469 views', 'short_view_count': '103K views'}        
{'video_id': 'jsp1cJAgS3I', 'title': 'Overwatch League 2021 Season | May Melee Tournament | Finals', 'view_count': '407,428 views', 'short_view_count': '407K views'}
{'video_id': 'eM38-PfPI0g', 'title': 'Overwatch League 2021 Season | May Melee Tournament | Day 2', 'view_count': '528,765 views', 'short_view_count': '528K views'}
{'video_id': 'zZZRYgycYF4', 'title': 'Overwatch League 2021 Season | May Melee Tournament | Day 1', 'view_count': '431,774 views', 'short_view_count': '431K views'}
{'video_id': 'ufQ35L4V-oU', 'title': 'Overwatch League 2021 Season | May Melee Qualifiers | Week 3 Day 4 — West', 'view_count': '643,822 views', 'short_view_count': '643K views'}
{'video_id': 'Q-wtjK3vyFo', 'title': 'Overwatch League 2021 Season | May Melee Qualifiers | Week 3 Day 3 — East', 'view_count': '331,533 views', 'short_view_count': '331K views'}
{'video_id': 'P6HEU-QOD0g', 'title': 'Overwatch League 2021 Season | May Melee Qualifiers | Week 3 Day 3 — West', 'view_count': '601,103 views', 'short_view_count': '601K views'}
{'video_id': 'xShlGPx6dZU', 'title': 'Overwatch League 2021 Season | May Melee Qualifiers | Week 3 Day 2 — East', 'view_count': '227,639 views', 'short_view_count': '227K views'}
{'video_id': 'xDnyQH5c4LI', 'title': 'Overwatch League 2021 Season | May Melee Qualifiers | Week 3 Day 2 — West', 'view_count': '495,053 views', 'short_view_count': '495K views'}
{'video_id': 'uQpLrumQP0E', 'title': 'Overwatch League 2021 Season | May Melee Qualifiers | Week 3 Day 1 — East', 'view_count': '260,034 views', 'short_view_count': '260K views'}
{'video_id': 'NRQjE6oJEak', 'title': 'Overwatch League 2021 Season | May Melee Qualifiers | Week 3 Day 1 — West', 'view_count': '370,308 views', 'short_view_count': '370K views'}
{'video_id': 'BaVMTjlnGZA', 'title': 'Overwatch League 2021 Season | May Melee Qualifiers | Week 2 Day 3 — West', 'view_count': '582,686 views', 'short_view_count': '582K views'}
63
```

**Comment by HuskyTrojan** [comment]
Thanks! I will try it out.

**Comment by xenova** [comment]
To differentiate between video_status and video_type, I've made some to the above changes:

`video_status` is one of: 'live', 'upcoming' or 'past'.
`video_type` is one of 'clip', 'premiere' or 'video'

(see https://github.com/xenova/chat-downloader/issues/106 and 6d0b260d10f266f33c9783832ec365c43e81fa12)

**TL DR** In the above examples, `video_type` will change to `video_status`

```python
from chat_downloader.sites import YouTubeChatDownloader

vids = YouTubeChatDownloader().get_user_videos(custom_username='overwatchleague', video_status='past')

count = 0
for video in vids:
    print(video)
    count += 1

print(count)
```

**Comment by xenova** [comment]
After further testing, these changes have now been included in the latest version (https://github.com/xenova/chat-downloader/releases/tag/v0.1.8)

You can install it using
```
pip install chat-downloader --upgrade
```

---

Let me know if you have any further questions 👍
